one-ui/src/main/java/com/marklogic/hub/oneui/controllers/SavedQueriesController.java

```diff
@@ -0,0 +1,47 @@
+package com.marklogic.hub.oneui.controllers;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.marklogic.hub.oneui.managers.SavedQueriesManager;
+import com.marklogic.hub.oneui.models.HubConfigSession;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Scope;
+import org.springframework.context.annotation.ScopedProxyMode;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+@Controller
+@RequestMapping(value = "/api/savedQueries")
+public class SavedQueriesController {
+
+    @Autowired
+    private SavedQueriesManager savedQueriesManager;
+
+    @Autowired
+    private HubConfigSession hubConfig;
+
+    @Bean
+    @Scope(proxyMode = ScopedProxyMode.TARGET_CLASS, value = "request")
+    SavedQueriesManager savedQueriesManager() {
+        return new SavedQueriesManager(hubConfig);
+    }
+
+    @RequestMapping(method = RequestMethod.POST)
+    @ResponseBody
+    public ResponseEntity<JsonNode> saveQueryDocument(@RequestBody JsonNode queryDocument) {
+        JsonNode savedQuery = savedQueriesManager.saveOrUpdateQueryDocument(queryDocument);
+        return new ResponseEntity<>(savedQuery, HttpStatus.CREATED);
+    }
+
+    @RequestMapping(method = RequestMethod.PUT)
+    @ResponseBody
+    public ResponseEntity<JsonNode> updateQueryDocument(@RequestBody JsonNode queryDocument) {
+        JsonNode savedQuery = savedQueriesManager.saveOrUpdateQueryDocument(queryDocument);
+        return new ResponseEntity<>(savedQuery, HttpStatus.OK);
```

Comment:
No need to change now, but there's a ResponseEntity.ok method you can use to get this all done in one line of code (I think by rule, a controller method should be one line of code, but there will of course be exceptions).

There's also ResponseEntity.created, but it requires a URI - which makes me think that perhaps the saveQueryDocument method should just return the queryId?